pandas/tests/io/parser/test_dtypes.py

```diff
@@ -509,3 +509,27 @@ def test_numeric_dtype(all_parsers, dtype):
 
     result = parser.read_csv(StringIO(data), header=None, dtype=dtype)
     tm.assert_frame_equal(expected, result)
+
+
+def test_intna_precision(all_parsers):
```

Comment:
Going to need several more tests to check this behavior, but we can get revisit this point after the actual implementation has been critiqued.